src/CiviProxy.php

```diff
@@ -0,0 +1,126 @@
+<?php
+/*--------------------------------------------------------+
+| SYSTOPIA CiviProxy                                      |
+|  a simple proxy solution for external access to CiviCRM |
+| Copyright (C) 2015-2021 SYSTOPIA                        |
+| Author: Jaap Jansma (jaap.jansma@civicoop.org           |
+| http://www.systopia.de/                                 |
++---------------------------------------------------------*/
+
+namespace systopia\CiviProxy;
+
+use systopia\CiviProxy\Api\Api;
+
+class CiviProxy {
+
+  /**
+   * @var CiviProxy
+   */
+  private static $instance;
+
+  private $plugins = [];
+
+  private $apiFactory;
+
+  private $eventListeners = [];
+
+  private $sortedListeners = [];
+
+  /**
+   * @return CiviProxy
+   */
+  public static function getInstance() {
+    if (!self::$instance) {
+      self::$instance = new CiviProxy();
+    }
+    return self::$instance;
+  }
+
+  private function __construct()
+  {
+    $this->initializePlugins();
+  }
+
+  private function initializePlugins() {
+    global $plugins;
+    foreach($plugins as $pluginClass) {
+      if (class_exists($pluginClass)) {
+        $plugin = new $pluginClass();
+        foreach ($plugin->getSubscribedEvents() as $eventName => $params) {
+          if (\is_string($params)) {
+            $this->addListener($eventName, [$plugin, $params]);
+          } elseif (\is_string($params[0])) {
+            $this->addListener($eventName, [$plugin, $params[0]], $params[1] ?? 0);
+          } else {
+            foreach ($params as $listener) {
+              $this->addListener($eventName, [$plugin, $listener[0]], $listener[1] ?? 0);
+            }
+          }
+        }
+        $this->plugins[] = $plugin;
+      }
+    }
+  }
+
+  public function addListener(string $eventName, $listener, int $priority = 0) {
+    $this->eventListeners[$eventName][$priority][] = $listener;
+    unset($this->sortedListeners[$eventName]);
+  }
+
+  public static function callApi($action) {
+    $civiproxy = CiviProxy::getInstance();
+    if (!$civiproxy->apiFactory) {
+      $civiproxy->apiFactory = new Api($civiproxy->plugins);
+    }
+    return $civiproxy->apiFactory->callApi($action);
+  }
+
+  /**
+   * @param object $event
+   */
+  public static function dispatchEvent($event) {
+    $eventName = get_class($event);
+
+    $civiproxy = CiviProxy::getInstance();
```

Comment:
Let's not make this method static (and use `$this` instead of calling `getInstance()`).